contract/src/main/kotlin/io/provenance/scope/loan/utility/ServicingDataFunctions.kt

```diff
@@ -0,0 +1,86 @@
+package io.provenance.scope.loan.utility
+
+import io.provenance.scope.util.toOffsetDateTime
+import tech.figure.servicing.v1beta1.LoanStateOuterClass.LoanStateMetadata
+import tech.figure.servicing.v1beta1.LoanStateOuterClass.ServicingData
+
+/* TODO: Adjust signatures and/or use Kotlin's new context receivers to make this file more digestable */
+
+internal fun updateServicingData(
+    existingServicingData: ServicingData = ServicingData.getDefaultInstance(),
+    newServicingData: ServicingData,
+): ServicingData = existingServicingData.toBuilder().also { servicingDataBuilder ->
+    validateRequirements(ContractRequirementType.VALID_INPUT) {
+        // TODO: Validate any other top-level fields of newServicingData?
+        // TODO: Which top-level fields of the servicing data record other than the loan states should be updatable? Does it vary between contracts?
```

Comment:
For now, I think we limit our contracts to appending loan states and appending doc metadata to this list. We will need a new contract for the latter - it is the list of docs that shows up in PM, which is a subset of the full docs list. We could eventually have one fancy append doc contract that lets users flag each doc for inclusion in PM, which would append it to both lists.

If in the long run we have a new use case to update the other high-level servicing data, then we will use a new contract.